vsphere/datadog_checks/vsphere/metadata_cache.py

```diff
@@ -41,6 +43,21 @@ def set_metadata(self, key, metadata):
         with self._metadata_lock:
             self._metadata[key] = metadata
 
+    def set_metric_ids(self, key, metric_ids):
+        """
+        Store the list of metric IDs we will want to collect for the given instance key
+        """
+        with self._metadata_lock:
```

Comment:
nit: we should probably rename the mutex to just `_lock` to better communicate that it now protects the whole cache, not only `self._metadata`